docs/sequence-diagrams/query.puml

```diff
@@ -0,0 +1,6 @@
+@startuml query
+client -> thanos_query: Request for data
```

Comment:
There is more to it, but it's a good start. 

There is also deduplication and filtering step.